dashboard/app/controllers/followers_controller.rb

```diff
@@ -65,35 +65,26 @@ def student_user_new
       return
     end
 
-    if current_user && @section
-      @section.add_student current_user
+    @user = current_user || User.new
 
-      redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name)
-    else
-      @user = User.new
-
-      # if this is a picture or word section, redirect to the section login page so that the student
-      # does not have to type in the full URL
-      if @section && [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD].include?(@section.login_type)
-        redirect_to controller: 'sections', action: 'show', id: @section.code
-      end
-
-      # if there is no logged in user and no section or an e-mail section, render the default
-      # student_user_new view which includes the section code form or sign up form
+    # if this is a picture or word section, redirect to the section login page so that the student
+    # does not have to type in the full URL
+    if @section && [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD].include?(@section.login_type)
+      redirect_to controller: 'sections', action: 'show', id: @section.code
     end
+
+    # render the default student_user_new view, which includes the section code form or sign up form
   end
 
   # POST /join/XXXXXX
   # join a section as a new student
```

Comment:
super tiny nit: I think this would be more accurate if it just said "join a section"